android/eir/src/main/java/org/smartregister/fhircore/eir/ui/patient/details/PatientDetailsFragment.kt

```diff
@@ -114,6 +118,44 @@ class PatientDetailsFragment private constructor() : Fragment() {
           )
       )
     }
+
+    reportAdverseEventButton.setOnClickListener {
+      val immunizations = patientDetailsViewModel.patientImmunizations.value as List<Immunization>
+      val immunizationItemWithIds =
+        immunizations.toImmunizationAdverseEventItem(requireContext()).first()
+
+      goToAdverseEventQuestionnaireActivity(immunizationItemWithIds, patientId)
+    }
+  }
+
+  private fun goToAdverseEventQuestionnaireActivity(
+    immunizationAdverseEventItem: ImmunizationAdverseEventItem,
+    patientId: String
+  ) {
+
+    val list = arrayListOf<String>()
+    immunizationAdverseEventItem.dosesWithAdverseEvents.forEach { dose -> list.add(dose.first) }
+
+    AlertDialog.Builder(requireActivity(), R.style.Theme_AppCompat_Light_Dialog_Alert)
+      .apply {
+        setTitle("Choose dose of ${immunizationAdverseEventItem.vaccine} for adverse events")
```

Comment:
do strings like this go into a text file for translation?

Reply:
I think, it should be, let me fix it